### PR TITLE
Add tests for Terraform::ResourceOverride

### PR DESCRIPTION
**Pull request by rosbo**
Change-Id: If688222244ff47b961090b0373bcb1fdb1685354

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: spec/provider_terraform_resource_override_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  end
+
+  def allow_open(file_name)
+    IO.expects(:read).with(file_name).returns(File.real_read(file_name))
+      .at_least(0)
+  end
+
+  def create_override(property_name, override_val)
+    Google::YamlValidator.parse(
+      format("--- !ruby/object:Provider::Terraform::ResourceOverride\n" \
```
[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: Done

File: spec/provider_terraform_resource_override_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    it 'overrides description' do
+      create_override('description', 'bar').apply resource
+
+      expect(resource.description).to eq 'bar'
+    end
+  end
+
+  def allow_open(file_name)
+    IO.expects(:read).with(file_name).returns(File.real_read(file_name))
+      .at_least(0)
```
[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: Done

File: spec/provider_terraform_resource_override_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    context 'with empty override' do
+      let(:override) { Provider::Terraform::ResourceOverride.new }
+      before { override.apply resource }
+
+      it { expect(resource.description).to eq 'foo' }
+    end
+
+    it 'extends description' do
+      create_override('description', '{{description}}bar').apply resource
```
[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: The subject is 'resource.description'. Where should I put create_override(...).apply if I am using subject?

File: spec/provider_terraform_resource_override_spec.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+describe Provider::Terraform::ResourceOverride do
+  context 'good resource' do
+    let(:resource) do
+      Provider::Terraform::ResourceOverride.parse(
+        IO.read('spec/data/good-resource.yaml')
+      )
+    end
+
+    before do
```
[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: Done

File: spec/provider_terraform_resource_override_spec.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      expect(resource.description).to eq 'foobar'
+    end
+
+    it 'overrides description' do
+      create_override('description', 'bar').apply resource
+
+      expect(resource.description).to eq 'bar'
+    end
+  end
+
```
[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: Done

File: spec/provider_terraform_resource_override_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (nelsonjr) [reply spec/provider_terraform_resource_override_spec.rb]: it seems the subject is always resource.description, so you can put the create_override in a before block.

[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: the subject is the same, but the create.override is different for each test. Creating a context block + a before block makes the test harder to read IMO

[the remark this answers is not in the document]
Comment (rosbo) [reply spec/provider_terraform_resource_override_spec.rb]: Done

**Pull request merged**